### PR TITLE
Make default template implementation cookiecutter

### PR DESCRIPTION
**Pull request by lucasmelin**
Update the `main` branch with the existing `cookiecutter` branch implementation. For now, the `cookiecutter` branch should not be deleted if this is merged in, since [python-project-template-repository](https://github.com/bocdaz/python-project-template-repository) and [fastapi-app-template-repository](https://github.com/bocdaz/fastapi-app-template-repository) currently specify the `cookiecutter` branch as the source.

Most of the changes proposed in this PR have been previously reviewed and approved in order to have been merged into the `cookiecutter` branch. The new commit as part of this PR is 2528cc5 which replaces `main` with the `cookiecutter` implementation, updates the README, and removes some files that are not needed for `cookiecutter`.

### CODE REVIEW

File: {{cookiecutter.hyphenated}}/src/{{cookiecutter.underscored}}/config.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    and then returned from cache for all subsequent calls.
+    """
+    return Settings()
+
+
+class Settings(BaseSettings):
+    """Shared Application Settings."""
+
+    # Settings for Security Dependencies
+    secret_key: str = "09d25e094faa6ca2556c818166b7a9563b93f7099f6f0f4caa6cf63b88e8d3e7"
```
Comment (slougheed) [inline {{cookiecutter.hyphenated}}/src/{{cookiecutter.underscored}}/config.py:25]: What is this secret_key used for? Should it be here and hardcoded?

**Review by slougheed — APPROVED** [review]
It looks fine to me, not sure about that "secret_key" but otherwise no issues.

File: {{cookiecutter.hyphenated}}/src/{{cookiecutter.underscored}}/config.py (hunk 1 above)
Comment (lucasmelin) [reply {{cookiecutter.hyphenated}}/src/{{cookiecutter.underscored}}/config.py:25]: It's an example used for signing JWTs, which is used for the `fake_users_db` in `dependencies.py`. It's necessary for the tests to work, but someone using the JWT implementation is supposed to swap out the `fake_users_db` and the `secret_key` used for demo purposes.

Comment (lucasmelin) [reply {{cookiecutter.hyphenated}}/src/{{cookiecutter.underscored}}/config.py:25]: It's the same secret used in the official docs https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/ (which should be a sign that this value is not a good secret to use in production).

**Pull request merged**